src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static Storage storage;
+    private User user1;
+    private User user2;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storage = new Storage();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user1 = new User();
+        user1.setLogin("Use1_Name");
+        user1.setPassword("password1");
+        user1.setAge(20);
+
+        user2 = new User();
+        user2.setLogin("User2_Name");
+        user2.setPassword("password2");
+        user2.setAge(21);
+    }
+
+    @AfterEach
+    void tearDown() {
+        storage.people.clear();
+    }
+
+    @Test
+    void putNullUse_Not_Ok() {
+        assertThrows(RuntimeException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void putNullLogin_Not_Ok() {
+        user1.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user1));
+    }
+
+    @Test
+    void putNonUniqueLogin_Not_Ok() {
+        user2.setLogin(user1.getLogin());
+        registrationService.register(user1);
```

Comment:
your "register" method may not work correctly. This is an excuse to add the first user directly to the "People List"
```suggestion
       storage.people.add(user1);
```

Reply:
I agree.